### DISCUSSION TITLE
Can I install OBS studio?

### DISCUSSION THREAD

**Discussion by ironiris** [Joshua-Riek/ubuntu-rockchip#28] [General]
I usually do YouTube streaming with my laptop and OBS software, but it would be great if I could do it on an SBC as well.

The 'apt install obs' command was installed without any errors, but when I try to run it, an error occurs.

**Comment by Joshua-Riek** [comment]
Someone has ported OBS for this board [here](https://launchpad.net/~liujianfeng1994/+archive/ubuntu/rockchip-multimedia), but I don't use the same drivers, so the install could break hardware acceleration.

However, I have been working on a solution; I recently got Kodi up and running from the same ppa and will create a tutorial soon. Using the same method, I should be able to get OBS working too. I can test this out and let you know the results and how to set it up later today.  I'm glad you brought this up, I have been working on increasing video playback performance for the next release, it will be interesting to see how OBS performs.

**Comment by ironiris** [comment]
Thank you for your hard work.

**Comment by Joshua-Riek** [comment]
I'm going to convert this issue to a discussion.

**Comment by Joshua-Riek** [comment]
Sorry for the late update; I was trying to get a release out tonight! I tested a few things by installing OBS, and it might be in a broken state. I keep getting errors saying that the GPU needs to be supported. We will need to wait for this to be fixed, or a workaround must be found.

**Comment by Joshua-Riek** [comment]
I was just able to get OBS working, it runs pretty well with a few tweaks! Ill document how to do this later tonight!
![Screenshot from 2023-02-26 23-53-56](https://user-images.githubusercontent.com/10427125/221445244-a03a9130-a70e-4f0b-b57e-1fef705ed74b.png)

**Comment by ironiris** [comment]
I've checked this news now.
Thank you for your hard work.
Additionally, Is it possible to check H/W encoding option?

**Comment by Joshua-Riek** [comment]
Yeah, it works with H/W encoding; the CPU was at around 8% usage during my test, and I still need to write up a guide on installing it. It's impressive to see the performance.

**Comment by Joshua-Riek** [comment]
## How to Install and Configure OBS Studio 
1. Add the below content to the file `/etc/apt/preferences.d/rockchip-multimedia-ppu`.
```
Package: *
Pin: release o=LP-PPA-liujianfeng1994-rockchip-multimedia
Pin-Priority: 10

Package: obs*
Pin: release o=LP-PPA-liujianfeng1994-rockchip-multimedia
Pin-Priority: 1001

Package: libobs*
Pin: release o=LP-PPA-liujianfeng1994-rockchip-multimedia
Pin-Priority: 1001
```
2. Add ppa and install packages
```
sudo add-apt-repository -y ppa:liujianfeng1994/rockchip-multimedia
sudo apt-get update
sudo apt-get install -y obs-studio obs-gstreamer qtwayland5
```
3. Add required environment variables 
```
echo "PAN_MESA_DEBUG=gl3" | sudo tee -a /etc/environment
echo "QT_QPA_PLATFORM=wayland" | sudo tee -a /etc/environment
source /etc/environment
```
4. Open OBS Stuido.
5. From within OBS Studio goto File -> Settings -> Output, and under the dropdown box labeled Output Mode, select Advanced.
6. Under the Streaming tab click the Encoder dropdown box and select GStreamer Encoder H.264.
7. Under the Recording tab click the Encoder dropdown box and select GStreamer Encoder H.265.
8. Click apply to save your changes.

**Reply by traxap** [reply]
Hello, 

Thank you in advance for your help.  I am new to ARM64 Linux but I'm learning new things everyday.  

I have not been able to reboot after trying to install and run OBS... with the above commands.  OPi5+ running off of a NVMe 1TB drive with KDE Plasma Wayland environment. 
Is there a way to restore to an earlier file system using Timeshift by accessing the hd on a windows PC? Any suggestions would be really appreciated.   
Thank you.  

Update:  
I was able to boot back into the system using a text editor and manually reversing all changes suggested above in the files:   "/etc/apt/preferences.d/rockchip-multimedia-ppu" and "/etc/environment" 
Steps: I removed the SSD from the OPi5+ and used an external USB to NVMe enclosure. Next, I plugged it into another PC that can read Linux files and edited these text files manually.

**Comment by ironiris** [comment]
I'm guessing it's probably an encoder issue.
(H.264 is fine, H.265 is problematic).
Your solution is close to perfect.
Thank you so much.

**Reply by hideyuki-o** [reply]
Hello
OBS-studio need qt6-base-dev qt6-wayland -15Kernel  , pleae install it.
sample recordinng file is
     https://drive.google.com/file/d/1lmyif-BLv1bSB2ZIgvryUDQiu3eZdmZJ/view?usp=drive_link
     
https://drive.google.com/file/d/1UTlbE8KLIgYarx4wD3DPETOqdXC6jZPr/view?usp=sharing

Thank you.

**Comment by c1ngular** [comment]
any success on this ? i would like to custom build obs studio on ubuntu-rockchip as well .

**Comment by Andy18650** [comment]
Is there an update to this discussion? I have successfully got OBS working with qtwayland6 without xcb or xwayland. However, there is no H/W encoder support. I have used the official apt repository (a mirror to be precise) instead of the PPA since I am a bit hesitant to include someone's PPA that has apparently caused someone's system to not boot. Is the PPA still the best method to get H/W encoders to work?

**Comment by matteoguglielmi** [comment]
@Joshua-Riek Hello Joshua, can you advise us on how to get hardware encoding with OBS and your 24.04 Ubuntu image? So far, I could install the latest version of OBS on my Orange Pi 5 Plus using the packages I found on [Pi-Apps](https://pi-apps.io/) as follows:

```
# purge ubuntu's stock obs-studio

dpkg -l | grep obs

aptitude purge libobs0t64 obs-plugins obs-studio

# get the latest obs-studio from pi-apps (find obs-studio 'noble' link in this script):
#
# https://github.com/Botspot/pi-apps/blob/master/apps/OBS%20Studio/install-64

wget https://github.com/Pi-Apps-Coders/files/releases/download/large-files/obs-studio-30.2.2-1-arm64-noble.deb

apt install ./obs-studio-30.2.2-1-arm64-noble.deb # this will install the required deps

apt-mark hold obs-studio

aptitude update

aptitude upgrade # make sure aptitude is not prompting you to DOWNGRADE obs-studio
```

Can we hardware accelerate it now?

Thank you.
